reminders.go

```diff
@@ -0,0 +1,82 @@
+package slack
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"time"
+)
+
+type Reminder struct {
+	ID         string    `json:"id"`
+	Creator    string    `json:"creator"`
+	User       string    `json:"user"`
+	Text       string    `json:"text"`
+	Recurring  bool      `json:"recurring"`
+	Time       time.Time `json:"time"`
+	CompleteTS int       `json:"complete_ts"`
+}
+
+type reminderResp struct {
+	SlackResponse
+	Reminder Reminder `json:"reminder"`
+}
+
+func (api *Client) doReminder(ctx context.Context, path string, values url.Values) (*Reminder, error) {
+	response := &reminderResp{}
+	if err := postSlackMethod(ctx, api.httpclient, path, values, response, api); err != nil {
+		return nil, err
+	}
+	if !response.Ok {
+		return nil, errors.New(response.Error)
+	}
+	return &response.Reminder, nil
+}
+
+// AddChannelReminder adds a reminder for a channel.
+//
+// See https://api.slack.com/methods/reminders.add (NOTE: the ability to set
+// reminders on a channel is currently undocumented but has been tested to
+// work)
+func (api *Client) AddChannelReminder(channelID, text, time string) (*Reminder, error) {
+	values := url.Values{
+		"token":   {api.token},
+		"text":    {text},
+		"time":    {time},
+		"channel": {channelID},
+	}
+	return api.doReminder(context.Background(), "reminders.add", values)
+}
+
+// AddUserReminder adds a reminder for a user.
+//
+// See https://api.slack.com/methods/reminders.add (NOTE: the ability to set
+// reminders on a channel is currently undocumented but has been tested to
+// work)
+func (api *Client) AddUserReminder(userID, text, time string) (*Reminder, error) {
```

Comment:
time as a string is a little unclear what is the expected format?

Reply:
It can be more than just a timestamp. According to [the docs](https://api.slack.com/methods/reminders.add) it is:

> When this reminder should happen: the Unix timestamp (up to five years from now), the number of seconds until the reminder (if within 24 hours), or a natural language description (Ex. "in 15 minutes," or "every Thursday")

So a string seems most fitting IMO.